### PR TITLE
[choreo] Add differential drivetrain support

### LINKED ISSUE #537: Add differential drive support
The TrajoptLib side was completed in https://github.com/SleipnirGroup/Choreo/pull/525.

* Extend the Choreo file format to choose between swerve and differential
* Make the Tauri backend call the appropriate functions for each (the TrajoptLib APIs for swerve and differential are very similar at least)
* Modify the frontend to take in differential-specific physical constants

The path UI should work the same except for the segment point-at constraint being unavailable for differential.

### PR DESCRIPTION
**Pull request by Superbro525Alt**
Resolves #537

### CODE REVIEW

File: src-tauri/src/main.rs [hunk 1]
```diff
@@ -286,8 +539,11 @@ async fn generate_trajectory(
     polygonObstacles: Vec<PolygonObstacle>,
     // The handle referring to this path for the solver state callback
     handle: i64,
-) -> Result<SwerveTrajectory, String> {
-    let mut path_builder = SwervePathBuilder::new();
+) -> TrajectoryExport {
+    create_builder!(config.tank);
+
+    println!("{:?}", config);
```
Comment (spacey-sooty) [inline src-tauri/src/main.rs:545]:
```suggestion
```

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+use core::panic;
```
Comment (spacey-sooty) [inline src-tauri/src/main.rs:4]:
```suggestion
```
You shouldn't import from core.
Shouldn't be required anyways

File: src-tauri/src/main.rs [hunk 3]
```diff
@@ -444,7 +716,9 @@ async fn generate_trajectory(
     for o in polygonObstacles {
         path_builder.sgmt_polygon_obstacle(0, wpt_cnt - 1, o.x, o.y, o.radius);
     }
+
```
Comment (spacey-sooty) [inline src-tauri/src/main.rs:719]:
```suggestion
```

File: src-tauri/src/main.rs [hunk 4]
```diff
@@ -444,7 +716,9 @@ async fn generate_trajectory(
...
     path_builder.set_drivetrain(&drivetrain);
+
```
Comment (spacey-sooty) [inline src-tauri/src/main.rs:721]:
```suggestion
```

**Review by spacey-sooty — CHANGES_REQUESTED** [review]

**Comment by shueja** [comment]
Superseded by other commits and UI rearchitecture. Thank you for the work though, since this will help us do the UI changes.

**Pull request closed without merging**